tools/integration-tests/tester/framework/network.go

```diff
@@ -0,0 +1,179 @@
+		panic(err)
+	}
+
+	n.entryNodeIdentity = identity.New(publicKey)
+	seed := base64.StdEncoding.EncodeToString(ed25519.PrivateKey(privateKey.Bytes()).Seed())
+
+	// create entry node container
+	n.entryNode = NewDockerContainer(n.dockerClient)
+	n.entryNode.CreateGoShimmerEntryNode(n.namePrefix(containerNameEntryNode), seed)
+	n.entryNode.ConnectToNetwork(n.id)
+	n.entryNode.Start()
+}
+
+// CreatePeer creates a new peer/GoShimmer node in the network and returns it.
+func (n *Network) CreatePeer() *Peer {
+	name := n.namePrefix(fmt.Sprintf("%s%d", containerNameReplica, len(n.peers)))
+
+	// create identity
+	publicKey, privateKey, err := hive_ed25519.GenerateKey()
+	if err != nil {
+		panic(err)
+	}
+	seed := base64.StdEncoding.EncodeToString(ed25519.PrivateKey(privateKey.Bytes()).Seed())
+
+	// create Docker container
+	container := NewDockerContainer(n.dockerClient)
+	container.CreateGoShimmerPeer(name, seed, n.namePrefix(containerNameEntryNode), n.entryNodePublicKey())
+	container.ConnectToNetwork(n.id)
+	container.Start()
+
+	peer := newPeer(name, identity.New(publicKey), container)
+	n.peers = append(n.peers, peer)
+	return peer
+}
+
+// Shutdown creates logs and removes network and containers.
+// Should always be called when a network is not needed anymore!
+func (n *Network) Shutdown() {
+	// stop containers
+	n.entryNode.Stop()
+	for _, p := range n.peers {
+		p.Stop()
+	}
+
+	// retrieve logs
+	createLogFile(n.namePrefix(containerNameEntryNode), n.entryNode.Logs())
+	for _, p := range n.peers {
+		createLogFile(p.name, p.Logs())
+	}
+
+	// remove containers
+	n.entryNode.Remove()
+	for _, p := range n.peers {
+		p.Remove()
+	}
+
+	// disconnect tester from network otherwise the network can't be removed
+	n.tester.DisconnectFromNetwork(n.id)
+
+	// remove network
+	err := n.dockerClient.NetworkRemove(context.Background(), n.id)
+	if err != nil {
+		panic(err)
+	}
+}
+
+// WaitForAutopeering waits until all peers have reached the minimum amount of neighbors.
+// Panics if this minimum is not reached after autopeeringMaxTries.
+func (n *Network) WaitForAutopeering(minimumNeighbors int) {
+	fmt.Printf("Waiting for autopeering...\n")
+	defer fmt.Printf("Waiting for autopeering... done\n")
+
+	maxTries := autopeeringMaxTries
+	for maxTries > 0 {
+
+		for _, p := range n.peers {
+			if resp, err := p.GetNeighbors(false); err != nil {
+				fmt.Printf("request error: %v\n", err)
+			} else {
+				p.SetNeighbors(resp.Chosen, resp.Accepted)
+			}
+		}
+
+		// verify neighbor requirement
+		min := 100
+		total := 0
+		for _, p := range n.peers {
+			neighbors := p.TotalNeighbors()
+			if neighbors < min {
+				min = neighbors
+			}
+			total += neighbors
+		}
+		if min >= minimumNeighbors {
+			fmt.Printf("Neighbors: min=%d avg=%.2f\n", min, float64(total)/float64(len(n.peers)))
+			return
+		}
+
+		fmt.Println("Not done yet. Try again in 5 seconds...")
+		time.Sleep(5 * time.Second)
+		maxTries--
+	}
+	panic("Peering not successful.")
```

Comment:
I was thinking about error handling in general. While developing I just put a panic everywhere to make it easier and faster to test. How should we handle errors emerging from Docker in general?

Reply:
I think everything in framework etc should rather return errors instead of panics. And then in the actual tests we would check for those with `require.noError` or something.
If someone uses the framework outside of tests, they are supposed to do their own error handling